redis/pubsub.go

```diff
@@ -60,27 +60,35 @@ func (c PubSubConn) Close() error {
 
 // Subscribe subscribes the connection to the specified channels.
 func (c PubSubConn) Subscribe(channel ...interface{}) error {
-	c.Conn.Send("SUBSCRIBE", channel...)
+	if err := c.Conn.Send("SUBSCRIBE", channel...); err != nil {
+		return err
+	}
```

Comment:
Changes like this seem quite dangerous. While this is the _correct_ behaviour, I do worry about the effect on applications using this package depending on the current behaviour of flush the connection regardless of the outcome of the send. Potentially this warrants a minor or even major version bump depending on how impactful this could be.

Reply:
My take on **this particular case** is that the `Flush` should be returning an error if the `Send` failed i.e. if `Send` failed `Flush` should also. Thoughts?

Reply:
Yeah, digging into it a bit more, conn is using a `bufio.Writer` for writes, whose behaviour described here https://golang.org/pkg/bufio/#Writer indicates that should a `Write` call to it fail (which is what `Send` will ultimately end up doing) then subsequent calls to `Flush` should also fail, so yeah, I think it's fine as it shouldn't actually result in a change in behaviour anyway.